cmd/template-resolver/utils/resolver_utils.go

```diff
@@ -291,28 +304,43 @@ func processConfigPolicyTemplate(
 	return nil
 }
 
-// ProcessObjTemplatesRaw takes a YAML string representation and
-// resolves the object's hub and managed templates
+// processObjTemplatesRaw takes a YAML string representation and resolves the object's managed templates
 func processObjTemplatesRaw(
 	raw *unstructured.Unstructured,
 	resolver *templates.TemplateResolver,
 ) error {
-	var rawDataList [][]byte
-
-	resolveOptions := templates.ResolveOptions{}
+	resolveOptions := templates.ResolveOptions{InputIsYAML: true}
 
 	oTRaw, _, _ := unstructured.NestedString(raw.Object, "object-templates-raw")
-	if oTRaw != "" {
-		resolveOptions.InputIsYAML = true
-
-		rawDataList = [][]byte{[]byte(oTRaw)}
-	} else {
+	if oTRaw == "" {
 		return fmt.Errorf("invalid object-templates-raw after resolving hub templates")
 	}
 
-	objectTemplates, err := resolveObjTemplates(true, rawDataList, resolver, resolveOptions)
+	if bytes.Contains([]byte(oTRaw), []byte("{{hub")) {
+		return fmt.Errorf("unresolved hub template in YAML input. Use the -hub-kubeconfig argument")
```

Comment:
This requires a `--hub-kubeconfig` instead of the single dash now that it's a Cobra CLI. This might be an issue elsewhere too.

Reply:
I did find it somewhere else, thanks for noticing!